src/index.ts

```diff
@@ -540,3 +550,77 @@ export const createAccountsMintsAndTokenAccounts = async (
     tokenAccounts,
   };
 };
+
+export const makeTokenMint = async (
+  connection: Connection,
+  mintAuthority: Keypair,
+  name: string,
+  symbol: string,
+  decimals: number,
+  uri: string,
+  additionalMetadata: Array<[string, string]> = [],
```

Comment:
The helper function is missing the `createUpdateFieldInstruction` that adds the `additionalMetadata` to the mint account. 

The `createInitializeInstruction` instruction only initializes some "standard" metadata fields (name, symbol, uri).

Each custom metadata field would require its own instruction.

For example:
```
        createUpdateFieldInstruction({
            metadata: mint.publicKey,
            updateAuthority: payer.publicKey,
            programId: TOKEN_2022_PROGRAM_ID,
            field: metadata.additionalMetadata[0][0],
            value: metadata.additionalMetadata[0][1],
        }),
```

https://github.com/solana-labs/solana-program-library/blob/master/token/js/examples/metadata.ts#L81-L87

One solution may be to loop through the `additionalMetadata` array and add an instruction for each key/value pair.  However, if there are many custom fields, it may require creating multiple transactions due to transaction size limit.

Reply:
Done - mapped key/value pairs instructions, also tweaked it a little to allow people to specify as an object if they prefer. 8 KV pairs is fine without hitting TX size limit.

Reply:
I'm closing this, but if you have feelings or ideas re: TX size limit feel free to reopen. I have hit same TX size limit with `createAccountsMintsAndTokenAccounts()` too. 

We could do multiple transactions, but are their other techniques that are still atomic?